CommuniTEAM/CommuniTEA
# [TEA-61] pgxpool / UUID Setup

### In this PR, I'm adding/changing the following:

- Fixed pre-commit issue on Windows where hooks would run twice
- Added additional SQLFluff hook to accommodate parsing issues (the "fix" hook refused to give line numbers, but the "lint" hook provides them)
- Implemented [pgxpool](https://pkg.go.dev/github.com/jackc/pgx/v5@v5.5.1/pgxpool) with a new file: `dbpool.go` in which the connection pool is established to allow for concurrent connections
   - db connections are now acquired from the pool instead of created per endpoint; existing endpoints were changed to match
- Added PUBLIC_URL to `.env.sample` (add it to your local `.env`!) to support seamless URL changes
    - Added a warning in the backend if the PUBLIC_URL isn't set in the local `.env`
- Altered the sqlc config to work with the json setup of our API endpoints: see the [sqlc documentation](https://docs.sqlc.dev/en/stable/reference/config.html#go) for more info on available configurations
- Added a new query and endpoint for locations_cities: POST a new city
- Added the [Google UUID](https://github.com/google/uuid) package

### Please do the following to test my code:

1. Add the PUBLIC_URL variable in `.env.sample` to your local `.env`
2. Delete and rebuild docker containers
3. Navigate to localhost:8000/docs and attempt to create a new city with the data:
```
{
  "name": "Portland",
  "state": "OR"
}
``` 
4. Ensure the response is successful and includes a UUID

[Related Jira Ticket](https://communiteam.atlassian.net/browse/TEA-61)
